src/Voron/Impl/Paging/RvnMemoryMapPager.cs

```diff
@@ -97,7 +97,7 @@ protected override bool CanPrefetchQuery()
         {
             var state = pagerState ?? _pagerState;
 
-            if (CanPrefetch.Value)
+            if (CanPrefetch.Value && _pagerState != null)
```

Comment:
Shouldn't we use `state`, instead of `_pageState` here? The value may be different, no?

Reply:
Also looked a bit weird to me but I didn't wanted to change it here. Asked @redknightlois to double check that.

Reply:
Federico just confirmed. I will create an issue and send a PR

Reply:
Fixed in https://github.com/ravendb/ravendb/pull/12160